### DISCUSSION TITLE
Display the current shell's name

### DISCUSSION THREAD

**Discussion by srid** [starship/starship#2441] [Q&A] [answered]
I use PowerShell as my default shell, but occasionally switch to bash (either manually or via `nix-shell`). How can I write a custom command that will display the current shell in use in the prompt?

This doesn't seem that straightforward to do in starship. You can't use `$SHELL` because that tells you the default shell, not current shell. `$0` is available only in bash.

**Comment by davidkna** [comment, marked as answer]
The next version will have a `shell` module for this purpose. In the meantime there are multiple solutions for this in the [custom module thread](https://github.com/starship/starship/discussions/1252).
